Implicit narrowing conversion in OffsetBitBlock

In Fuchsia, implicit narrowing conversions are errors, so OffsetBitBlock fails to compile:
```
../../third_party/github.com/google/emboss/src/runtime/cpp/emboss_memory_util.h:917:9: error: implicit conversion loses integer precision: 'int' to 'ValueType' (aka 'unsigned short') [-Werror,-Wimplicit-int-conversion]
        ~(MaskToNBits(static_cast<ValueType>(~ValueType{0}), size_) << offset_);

```
I'll work on a merge request.

--- a/runtime/cpp/emboss_memory_util.h
+++ b/runtime/cpp/emboss_memory_util.h
@@ -913,9 +913,10 @@ class OffsetBitBlock final {
 
  private:
   ValueType MaskInValue(ValueType original_value, ValueType new_value) const {
-    ValueType original_mask =
-        ~(MaskToNBits(static_cast<ValueType>(~ValueType{0}), size_) << offset_);
-    return (original_value & original_mask) | (new_value << offset_);
+    ValueType original_mask = static_cast<ValueType>(~(
+        MaskToNBits(static_cast<ValueType>(~ValueType{0}), size_) << offset_));
+    return static_cast<ValueType>((original_value & original_mask) |
+                                  (new_value << offset_));
   }
 
   const UnderlyingBitBlockType bit_block_;

